File: erp-frontend/src/api/material/order/model/queryMaterialOrderBo.ts
export interface QueryMaterialOrderBo {
  /**
   * ID
   */
  id: string;

  /**
   * 发料单号
   */
  code: string;

  /**
   * 仓库ID
   */
  scId: string;

  /**
   * 仓库名称
   */
  scName: string;

  /**
   * 发料申请单ID
   */
  materialApplyId?: string;

  /**
   * 发料申请单号
   */
  materialApplyCode?: string;

  /**
   * 总数量
   */
  totalNum: number;

  /**
   * 已出库数量
   */
  totalOutNum: number;

  /**
   * 总金额
   */
  totalAmount: number;

  /**
   * 备注
   */
  description?: string;

  /**
   * 创建人
   */
  createBy: string;

  /**
   * 创建时间
   */
  createTime: string;

  /**
   * 合同编号
   */
  contractCode?: string;

  /**
   * 合同名称
   */
  contractName?: string;

  /**
   * 客户标识
   */
  customerCode?: string;

  /**
   * 客户名称
   */
  customerName?: string;

  /**
   * 机型编号
   */
  machineTypeCode?: string;

  /**
   * 机型名称
   */
  machineTypeName?: string;

  /**
   * 件号编号
   */
  partNumberCode?: string;

  /**
   * 件号名称
   */
  partNumberName?: string;

  /**
   * 是否已完成出库
   */
  isOutFinish: boolean;
}

File: erp-backend/xingyun-shkb/src/main/java/com/lframework/xingyun/shkb/bo/material/QueryMaterialOrderBo.java
package com.lframework.xingyun.shkb.bo.material;

import com.lframework.starter.web.core.bo.BaseBo;
import com.lframework.xingyun.shkb.entity.MaterialOrder;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 发料单列表查询Bo
 *
 * @author kison
 */
@Data
public class QueryMaterialOrderBo extends BaseBo<MaterialOrder> {

    /**
     * ID
     */
    @ApiModelProperty("ID")
    private String id;

    /**
     * 单号
     */
    @ApiModelProperty("单号")
    private String code;

    /**
     * 仓库ID
     */
    @ApiModelProperty("仓库ID")
    private String scId;

    /**
     * 仓库名称
     */
    @ApiModelProperty("仓库名称")
    private String scName;

    /**
     * 发料数量
     */
    @ApiModelProperty("发料数量")
    private Integer totalNum;

    /**
     * 已发料数量
     */
    @ApiModelProperty("已发料数量")
    private Integer totalOutNum;

    /**
     * 发料金额
     */
    @ApiModelProperty("发料金额")
    private BigDecimal totalAmount;

    /**
     * 备注
     */
    @ApiModelProperty("备注")
    private String description;

    /**
     * 创建人
     */
    @ApiModelProperty("创建人")
    private String createBy;

    /**
     * 创建时间
     */
    @ApiModelProperty("创建时间")
    private LocalDateTime createTime;

    /**
     * 发料申请单ID
     */
    @ApiModelProperty("发料申请单ID")
    private String materialApplyId;

    /**
     * 发料申请单编号
     */
    @ApiModelProperty("发料申请单编号")
    private String materialApplyCode;

    /**
     * 合同编号
     */
    @ApiModelProperty("合同编号")
    private String contractCode;

    /**
     * 合同名称
     */
    @ApiModelProperty("合同名称")
    private String contractName;

    /**
     * 客户编号
     */
    @ApiModelProperty("客户编号")
    private String customerCode;

    /**
     * 客户名称
     */
    @ApiModelProperty("客户名称")
    private String customerName;

    /**
     * 机型编号
     */
    @ApiModelProperty("机型编号")
    private String machineTypeCode;

    /**
     * 机型名称
     */
    @ApiModelProperty("机型名称")
    private String machineTypeName;

    /**
     * 件号编号
     */
    @ApiModelProperty("件号编号")
    private String partNumberCode;

    /**
     * 件号名称
     */
    @ApiModelProperty("件号名称")
    private String partNumberName;
    
    /**
     * 是否已全部出库完毕
     */
    @ApiModelProperty("是否已全部出库完毕")
    private Boolean isOutFinish;

    public QueryMaterialOrderBo() {
    }

    public QueryMaterialOrderBo(MaterialOrder dto) {
        super(dto);
    }
    
    // 提供兼容的访问器方法
    public Boolean getIsOutFinish() {
        return isOutFinish;
    }
}
